tests/integration/test_valid_data_in_request.py

```diff
@@ -62,9 +62,15 @@ def test_npm_basic(test_env, default_requests):
 
     for item in response.data["dependencies"]:
         if item["name"] not in test_env["get"]["npm"]["non_dev_dependencies"]:
-            assert item["dev"]
+            assert item["dev"], (
+                f"#{response.id}: dependency {item['name']} is not a dev dependency,"
+                f" but it is not listed as non-dev dependency"
+            )
         else:
-            assert not item["dev"]
+            assert not item["dev"], (
+                f"#{response.id}: dependency {item['name']} is listed as non-dev dependency,"
+                f" but it is a dev dependency"
+            )
```

Comment:
IMHO these messages could be clearer about what is expected vs. actual, maybe something like `dependency {dep} should be listed as a dev dependency, but is listed as a non-dev dependency`

Reply:
It's hard for me to write these two messages so it makes sense for someone else. We only use `non_dev_dependencies` list in test data, so you're not listing dev dependencies anywhere. I'll try to come up with something better.

Reply:
changed it, what do you think?

Reply:
Hmm, I might have misunderstood what "listed" means. If something is listed as a non-dev dependency, it means that it is in the `non_dev_dependencies` list in the test data? The way the error messages are phrased now, it sounds like you should change the test expectations if there is an error :D

Maybe this would work?
`dependency {item['name']} is not listed as non-dev in the test expectations, but is listed as non-dev in the response`
`dependency {item['name']} is listed as non-dev in the test expectations, but is not listed as non-dev in the response`

Reply:
oh, that's true :facepalm: Your suggestion looks great.